engine/src/main/java/io/camunda/zeebe/engine/processing/common/ExpressionProcessor.java

```diff
@@ -41,6 +42,18 @@ public ExpressionProcessor(
     evaluationContext = new VariableStateEvaluationContext(lookup);
   }
 
+  /**
+   * Returns a new expression processor that will use {@code secondaryLookup} when the first lookup
+   * failed
+   *
+   * @param secondaryLookup additional lookup that comes into play when the first lookup failed
+   * @return expression processor that will use {@code secondaryLookup} when the first lookup failed
+   */
+  public ExpressionProcessor chain(final VariablesLookup secondaryLookup) {
+    return new ExpressionProcessor(
+        expressionLanguage, evaluationContext.lookup.chain(secondaryLookup));
+  }
```

Comment:
👍  I really like this approach

🔧  The naming makes it sound like we can chain this multiple times, but on first glance the implementation seems to override earlier calls to chain. Or am I perhaps reading this wrong?

🔧  It would be great if the API allows defining whether the provided variable lookup should be used before or after the other lookups chained so far. So a `chainBefore`, and `chainAfter` method or so. What do you think?

Reply:
agreed on method names for different ways to chain